sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -116,11 +116,15 @@ abstract class Attribute extends LeafExpression with NamedExpression {
  *
  * @param child the computation being performed
  * @param name the name to be associated with the result of computing [[child]].
+ * @param isGenerated a flag to indicate if this alias is generated by Catalyst
  * @param exprId A globally unique id used to check if an [[AttributeReference]] refers to this
  *               alias. Auto-assigned if left blank.
+ * @param qualifiers a list of strings that can be used to referred to this attribute in a fully
+ *                   qualified way. Consider the examples tableName.name, subQueryAlias.name.
+ *                   tableName and subQueryAlias are possible qualifiers.
  * @param explicitMetadata Explicit metadata associated with this alias that overwrites child's.
  */
-case class Alias(child: Expression, name: String)(
+case class Alias(child: Expression, name: String, isGenerated: Boolean = false)(
```

Comment:
Why not put `isGenerated` into the second list with the other metadata so that we don't have to rewrite every match statement in the code base?